asterius/src/Asterius/CodeGen.hs

```diff
@@ -829,11 +832,31 @@ marshalCmmPrimCall (GHC.MO_Memcmp _) [_cres] [_ptr1, _ptr2, _n] = do
         }
     ]
 marshalCmmPrimCall (GHC.MO_PopCnt GHC.W64) [r] [x] =
-  marshalCmmUnPrimCall PopcntInt64 I64 r x
+  marshalCmmUnPrimCall PopcntInt64 I64 r x id
+marshalCmmPrimCall (GHC.MO_PopCnt GHC.W32) [r] [x] = do
```

Comment:
Is it possible to use `PopcntInt32` for the `W32` case? Same for `Clz` and `Ctz`.

Reply:
I've done it but it seems to make the two tests fail again. I will try with a fresh build.

Reply:
Ok fixed.